src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC, useEffect, useRef, ChangeEventHandler, useState } from 'react';
+import { classNames } from '../../utils/classNames';
+import Icon from '../Icon';
+
+export type checkboxType = | 'primary' | 'secondary' | 'default';
+
+export type checkboxSize = | 'md' | 'sm';
+
+export interface ICheckboxProps {
+    className?: string;
+    checkSize?: checkboxSize;  // ? gives user option to provide this attribute or not
+    checkType?: checkboxType;
+    label?: string;
+    disabled?: boolean;
+    checkedState?: boolean;
+    icon?: string;
+    onChange?: ChangeEventHandler<HTMLInputElement>;
+    indeterminate?: boolean;
+}
+
+const Checkbox: FC<ICheckboxProps> = (props) => {
+    const {checkSize, checkType, label, disabled, checkedState, className, icon, onChange, indeterminate} = props;
+    const checkRef = useRef<any>(); // maintains reference to input so can manage indeterminate property directly
+    useEffect(() => { // upon render set input to checked if indeterminate        
+        checkRef.current.indeterminate = indeterminate === true;
+    }, [indeterminate]);
+
+    let styleClasses = classNames('checkbox', {
+        [`check-${checkType}`]: true,
+        [`check-${checkSize}`]: !!checkSize
+    });
+    if (className) {
+        styleClasses += ' ' + className;
+    }
+
+    let checkbox;
+    if (icon) {
+        // console.log(icon);
+        checkbox = (
+            <label className={styleClasses}>
+                <input type='checkbox' disabled={disabled} ref={checkRef} checked={checkedState} onChange={onChange} />
```

Comment:
I just wondering what happens if the user does not pass any checked value and onChange function, does the checkbox still working? I just think the react way, the checkbox should control by the state. I think it should work by default because the default behavior of the checkbox, react is not able to control it at this point

Reply:
I believe so.  If you look at my stories file, line 15, I have a checkbox with no checked value or onChange and it still works.  Is that what you meant?

Reply:
Well, this is the part where both of us have different options, inside my component I control the check stated with react state, and you can check the code I implemented by useState inside my component.

But When I check your code, I don't see anything related to the check state being controlled by the component state instead it's controlled by the props.

Reply:
but over all everything looks good so far